### ISSUE TITLE
Idea: Empty output if no matches

### ISSUE THREAD

**Issue by aude** [jhawthorn/fzy#105]
Currently, if I pipe in some lines to `fzy`, input the garble `oSh3ahngah1e` and press `Enter`, this happens:

```
$ ls | fzy
oSh3ahngah1e
```

I think I would prefer no output in case there is no match:

```
$ ls | fzy

```

I would be easier to use together with a Bash widget I use, that opens a fuzzy find file dialog via `rg` and `fzy`. Then I'd rather get no output than incorrect output for the filename.

What do you think?

**Comment by iamleot** [comment]
Hello Asbjørn, 

Asbjørn Apeland writes:
> Currently, if I pipe in some lines to `fzy`, input the garble `oSh3ahngah1e` and press `Enter`, this happens:
>
> ```
> $ ls | fzy
> oSh3ahngah1e
> ```
>
> I think I would prefer no output in case there is no match:
>
> ```
> $ ls | fzy
>
> ```
>
> I would be easier to use together with a Bash widget I use, that opens a fuzzy find file dialog via `rg` and `fzy`. Then I'd rather get no output than incorrect output for the filename.
>
> What do you think?
> [...]

I would find it a bit strange because it would limit the fzy output
to its piped input instead of giving the freedom to type and output
a possible user's input.

I like the dmenu behaviour where:

 - `Return' confirms the current selection
 - `Shift-Return' confirm the input

E.g.:

 % printf "foo\nfooo\nbar\nbaz\n" | dmenu

Typing `f' and pressing `Return' will confirm `foo' (the first
selection), like fzy.
Typing `f' and pressing `Shift-Return' will just output `f'.

**Comment by aude** [comment]
That makes sense to me.

The point I am interested in is, in your example, what happens if I type `c`? I would want empty output, as there is no selection (because nothing matches `c`).

What do you think about that?

**Comment by iamleot** [comment]
Asbjørn Apeland writes:
> [...]
> That makes sense to me.
>
> The point I am interested in is, in your example, what happens if I type `c`? I would want empty output, as there is no selection (because nothing matches `c`).
>
> What do you think about that?
> [...]

In dmenu (both if pressing `Return' or if pressing `Shift+Return')
the same of what happens in fzy: `c' is printed out.

**Comment by aude** [comment]
That is true, that's what happens in dmenu.

I would personally like if it behaved differently. Is that a possibility, or do you think it will not happen?

**Comment by jhawthorn** [comment]
Good idea. I would accept a PR which added `--only-exact` (or something similar) but I'd like to keep the default behaviour.

**Comment by zvezdochiot** [comment]
@jhawthorn say:
> I would accept a PR which added --only-exact

They forgot to mention the use of fzy, which requires an exact match:
```sh
cd /usr/bin
$(ls | fzy)
```
`--only-exact` is needed.

**Comment by Julian** [comment]
(This I suppose is independent, so if needed I can create a new one?) but I'd equally well love a way to have `fzy` automatically exit if there is *one* input.

I.e., the "combined" behavior is -- zero input, exit immediately. One input, immediately select it and write it to stdout. Any more, show the normal fuzzy matching.
